utils/typename/src/lib.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+pub use typename_derive::TypeName;
+
+/// The interface for getting a name of the type.
+///
+/// The name is used in debug output, so it shouldn't be too verbose (fully qualified) to nut
+/// clutter logs.
+pub trait TypeName {
+    /// Returns a name of the type.
+    fn typename_str() -> &'static str;
```

Comment:
Is there a reason this is a `fn` rather than `const`, given generics are not supported?

Reply:
No reason other than that I'm basing this on the TypeName crate. I'm not sure whether it's possible to be honest given that there's some macro mess going on there. Not my biggest point of strength.

Reply:
We changed this to Cow to support generics.